frontend/src/reducers/user.js

```diff
@@ -0,0 +1,145 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  login: {
+    accessToken: localStorage.accessToken || null,
+    userId: localStorage.userId || 0,
+    secretMessage: null,
+    errorMessage: null
+  }
+};
+
+export const user = createSlice({
+  name: 'user',
+  initialState: initialState,
+  reducers: {
+    setAccessToken: (state, action) => {
+      const { accessToken } = action.payload;
+      state.login.accessToken = accessToken;
+      localStorage.setItem('accessToken', accessToken);
+    },
+    setUserId: (state, action) => {
+      const { userId } = action.payload;
+      state.login.userId = userId;
+      localStorage.setItem('userId', userId);
+    },
+    setSecretMessage: (state, action) => {
+      const { secretMessage } = action.payload;
+      state.login.secretMessage = secretMessage;
+    },
+    setErrorMessage: (state, action) => {
+      const { errorMessage } = action.payload;
+      state.login.errorMessage = errorMessage;
+    }
+  }
+});
+
+// Thunks
+// Signup
+export const signup = (name, email, password) => {
+  // const SIGNUP_URL = 'http://localhost:8080/signup';
+  const SIGNUP_URL = 'https://auth-by-ylva-tara.herokuapp.com/signup';
+  return (dispatch) => {
+    fetch(SIGNUP_URL, {
+      method: 'POST',
+      body: JSON.stringify({ name, email, password }),
+      headers: { 'Content-Type': 'application/json' }
+    })
+      .then((res) => {
+        if (!res.ok /* if not 200, 201, 204 */) {
+          throw 'Could not create account. Email already exists.';
+        }
+
+        // OK
+        return res.json();
+      })
+      .then((json) => {
+        dispatch(
+          user.actions.setAccessToken({
+            accessToken: json.accessToken
+          })
+        );
+        dispatch(user.actions.setUserId({ userId: json.userId }));
+        dispatch(user.actions.setErrorMessage({ errorMessage: null }));
+      })
+      .catch((err) => {
+        dispatch(user.actions.setErrorMessage({ errorMessage: err }));
+      });
+  };
+};
+
+// Login
+export const login = (email, password) => {
+  // const LOGIN_URL = 'http://localhost:8080/login';
```

Comment:
The old URL could be removed from these lines

Reply:
Have removed this URLs.